app/helpers/application_helper.rb

```diff
@@ -1,2 +1,13 @@
 module ApplicationHelper
+  def anchor_id_for_field object, field
+    [
+      object.class.name,
+      field,
+    ].map do |s|
+      s.to_s
+       .gsub(" ", "_")
+       .gsub(/\W/, '')
+       .downcase
+    end.join("_")
+  end
```

Comment:
Would be nice to find a way to reuse the original logic that generates HTML `id`s 
https://github.com/rails/rails/blob/7f18ea14c893cb5c9f04d4fda9661126758332b5/actionview/lib/action_view/helpers/tags/base.rb#L99-L106